openrtb3/openrtb3.go

```diff
@@ -0,0 +1,162 @@
+	Ext    json.RawMessage `json:"ext,omitempty"`    // Optional exchange-specific extensions.
+}
+
+// ----------------------------------------------------------------------------
+
+// Response object is the bid response object under the Openrtb root.
+// Its id attribute is a reflection of the bid request ID.
+// The bidid attribute is an optional response tracking ID for bidders.
+// If specified, it will be available for use in substitution macros placed in markup and notification URLs.
+// At least one Seatbid object is required, which contains at least one Bid for an item.
+// Other attributes are optional.
+//
+// To express a "no-bid", the most compact option is simply to return an empty response with HTTP 204.
+// However, if the bidder wishes to convey a reason for not bidding, a Response object can be returned with just a reason code in the nbr attribute.
+type Response struct {
+	ID      string          `json:"id"`                // Required. ID of the bid request to which this is a response; must match the "request.id" attribute.
+	BidID   string          `json:"bidid,omitempty"`   // Bidder generated response ID to assist with logging/tracking.
+	NBR     NoBidReasonCode `json:"nbr,omitempty"`     // Reason for not bidding if applicable (see List: No-Bid Reason Codes). Note that while many exchanges prefer a simple HTTP 204 response to indicate a no-bid, responses indicating a reason code can be useful in debugging scenarios.
+	Cur     string          `json:"cur,omitempty"`     // Default: "USD". Bid currency using ISO-4217 alpha codes.
+	CData   string          `json:"cdata,omitempty"`   // Allows bidder to set data in the exchange’s cookie, which can be retrieved on bid requests (refer to “cdata” in Object: Request) if supported by the exchange. The string must be in base85 cookie-safe characters.
+	SeatBid []SeatBid       `json:"seatbid,omitempty"` // Array of “Seatbid” objects; 1+ required if a bid is to be made.
+	Ext     json.RawMessage `json:"ext,omitempty"`     //	Optional demand source specific extensions.
+}
+
+// SeatBid object.
+//
+// A bid response can contain multiple Seatbid objects, each on behalf of a different buyer seat and each containing one or more individual bids.
+// If multiple items are presented in the request offer, the package attribute can be used to specify if a seat is willing to accept any impressions that it can win (default) or if it is interested in winning any only if it can win them all as a group.
+type SeatBid struct {
+	Seat    string          `json:"seat,omitempty"`    // Recommended. ID of the buyer seat on whose behalf this bid is made.
+	Package int8            `json:"package,omitempty"` // Default: 0. For offers with multiple items, this flag Indicates if the bidder is willing to accept wins on a subset of bids or requires the full group as a package, where 0 = individual wins accepted; 1 = package win or loss only.
```

Comment:
Don't think there's any sense in making this pointer, as it defaults to `0` - so better save traffic instead of passing `0` around.